### ISSUE TITLE
Rule request: fallthrough should be avoided

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#1834]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

Avoid `fallthrough` inside `switch` statements.

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Using `fallthrough` makes code harder to read and, in most cases, can be replaced by combining `cases`. [LinkedIn's Swift Style Guide](https://github.com/linkedin/swift-style-guide#32-access-modifiers) also thinks that (item 3.4.5).

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// should trigger
switch foo {
case .bar:
    fallthrough
case .bar2:
    fallthrough
case .bar3
    something()
}


// shouldn't trigger
switch foo {
case .bar, .bar2, .bar3:
    something()
}
```
3. Should the rule be configurable, if so what parameters should be configurable?

Only severity.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Opt-in. In the end, this is still a matter of style and there are situations where you can't just combine cases.
